src/internal/obj/api.go

```diff
@@ -0,0 +1,31 @@
+package obj
+
+import (
+	"context"
+	"io"
+)
+
+// Client is an interface to object storage.
+type Client interface {
+	// Put writes the data from r to an object at name
+	// It should error if the object already exists or we don't have sufficient
+	// permissions to write it.
+	Put(ctx context.Context, name string, r io.Reader) error
+
+	// Get writes the data for an object to w
+	// If `size == 0`, the reader should read from the offset till the end of the object.
+	// It should error if the object doesn't exist or we don't have sufficient
+	// permission to read it.
+	Get(ctx context.Context, name string, offset uint64, size uint64, w io.Writer) error
```

Comment:
I think we should remove the offset / size parameters. We may want this as a functional option later, but I kind of think we shouldn't have this because I cannot think of a good use case for it. Also, it requires more from object storage providers and / or client implementations.

Reply:
I think this will be difficult to add back.  Were we ever using this? Are there any use cases coming up?

Reply:
I've thought about this more. I think it might become useful for PutFile from object storage.  It is easy to take out if we ever want to, but definitely more work to put back in.  I think we should keep it in for now.

Reply:
I disagree. I doubt it would be that useful for PutFile from object storage, but even if we still want to support it for that, the offset and size should not be required parameters for the interface. The default should be a full read and the size 0 being a full read shouldn't be a thing.